pkg/connector/projects.go

```diff
@@ -0,0 +1,131 @@
+package connector
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"cloud.google.com/go/bigquery"
+	resourcemanager "cloud.google.com/go/resourcemanager/apiv3"
+	"cloud.google.com/go/resourcemanager/apiv3/resourcemanagerpb"
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/pagination"
+	rs "github.com/conductorone/baton-sdk/pkg/types/resource"
+)
+
+type projectBuilder struct {
+	resourceType   *v2.ResourceType
+	projectsClient *resourcemanager.ProjectsClient
+	bigQueryClient *bigquery.Client
+}
+
+func (p *projectBuilder) ResourceType(ctx context.Context) *v2.ResourceType {
+	return projectResourceType
+}
+
+func projectResource(projects *resourcemanagerpb.Project) (*v2.Resource, error) {
+	var opts []rs.ResourceOption
+	profile := map[string]interface{}{
+		"id":          projects.ProjectId,
+		"name":        projects.Name,
+		"displayName": projects.DisplayName,
+	}
+
+	projectTraitOptions := []rs.AppTraitOption{
+		rs.WithAppProfile(profile),
+	}
+
+	opts = append(opts, rs.WithAppTrait(projectTraitOptions...))
+	resource, err := rs.NewResource(
+		projects.DisplayName,
+		projectResourceType,
+		projects.ProjectId,
+		opts...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return resource, nil
+}
+
+func (p *projectBuilder) List(ctx context.Context, parentResourceID *v2.ResourceId, pToken *pagination.Token) ([]*v2.Resource, string, annotations.Annotations, error) {
+	var resources []*v2.Resource
+	it := p.projectsClient.SearchProjects(ctx,
+		&resourcemanagerpb.SearchProjectsRequest{
+			Query: "",
+		},
+	)
+	_, bag, err := unmarshalSkipToken(pToken)
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	if bag.Current() == nil {
+		bag.Push(pagination.PageState{
+			ResourceTypeID: projectResourceType.Id,
+		})
+	}
+
+	project, err := it.Next()
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	err = bag.Next(project.ProjectId)
```

Comment:
ProjectId is a string, but unmarshalSkipToken() tries to parse it as an integer. This means that syncing projects always fails unless the project ID happens to only contain numbers.

Reply:
Fixed